### PR TITLE
automated MM-14632 Integration test for removing team icon #4066

### PR DESCRIPTION
**Pull request by VishalSwarnkar**
#### Related Pull Requests

- Has webapp changes (https://github.com/mattermost/mattermost-webapp/pull/4611/commits/05fb09b4eb642205ef5691280721a2b39814e1be)

### CODE REVIEW

**Comment by VishalSwarnkar** [comment]
@jasonblais, kindly review my PR, please.

**Comment by jasonblais** [comment]
Thanks @VishalSwarnkar! Would you be open to signing the above contributor license agreement, standard in open source projects? If you have any questions, don't hesitate to reach out.

Once signed, we'll queue your PR for reviews. Big thanks for your contribution!

**Comment by VishalSwarnkar** [comment]
> Thanks @VishalSwarnkar! Would you be open to signing the above contributor license agreement, standard in open source projects? If you have any questions, don't hesitate to reach out.
> 
> Once signed, we'll queue your PR for reviews. Big thanks for your contribution!

**Comment by VishalSwarnkar** [comment]
Thanks, @jasonblais I have signed and submitted CLA

**Comment by jasonblais** [comment]
Thanks @VishalSwarnkar, I've queued it for review. Would you be open to updating the PR description with a summary and ticket link?

**Comment by VishalSwarnkar** [comment]
Thanks, @reflog for reviewing the code. As suggested I have replaced newly introduced _id_ with _data-testid_ and updated the PR. Kindly review the changes

**Comment by VishalSwarnkar** [comment]
Hi @jasonblais the PR is now approved now but still, I can see the current status as "merge/blocked Expected — Waiting for status to be reported". Is there anything I have to do in regards to the PR?

File: e2e/cypress/integration/team/remove_team_icon_spec.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        openTeamSettingsDialog();
+
+        // verify the settings picture button is visible to click
+        cy.get('[data-testid="inputSettingPictureButton"]').should('be.visible').click();
+
+        // # Upload a file on center view
+        cy.fileUpload('[data-testid="uploadPicture"]', 'mattermost-icon.png');
+
+        // after uploading the picture the save button must be disabled
+        cy.get('[data-testid="saveSettingPicture"]').should('not.be.disabled').click();
```
Comment (saturninoabril) [inline e2e/cypress/integration/team/remove_team_icon_spec.js]: This is not accurate. Currently, the save button is always `disable: false`.  To check enable/disable state of the button, I'd suggest to add ``disabled={disableSaveButtonFocus}`` (or change the variable to make it ``disabled={disableSaveButton}``) [here](https://github.com/mattermost/mattermost-webapp/pull/4611/files#diff-bf2e0895a3d11fd13d4f4a4f34feb18bR342), then check enable/disable before/after file upload.

File: e2e/cypress/integration/team/remove_team_icon_spec.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            });
+        });
+    });
+
+    it('TS14632 Remove Team Icon', () => {
+        // function to open the team settings dialog
+        openTeamSettingsDialog();
+
+        // verify the settings picture button is visible to click
+        cy.get('[data-testid="inputSettingPictureButton"]').should('be.visible').click();
```
[the remark this answers is not in the document]
Comment (VishalSwarnkar) [reply e2e/cypress/integration/team/remove_team_icon_spec.js]: Hi @saturninoabril, thanks for the suggestion but as per @reflog review comments I have followed the https://docs.cypress.io/guides/references/best-practices.html and update the _ids_ to _data-testid_
Looking at the existing code base I can see the mix approach few of the specs has cy.findByTestId and few of them are using cy.get('[data-testid="xyz"]'). 
It would be really better if the approaches are consistent across, but please advise which approach is better?

[the remark this answers is not in the document]
Comment (saturninoabril) [reply e2e/cypress/integration/team/remove_team_icon_spec.js]: It's not a contrary to his comment.  Since you're using `data-testid`, the convenience function is to get it by `cy.findByTestId`.  Those using ``cy.get('[data-testid="xyz"]')`` works too but since we have ``cy.findByTestId``, let's use it.
We can fix others in separate PR to make it consistent all throughout the codebase.

**Comment by jasonblais** [comment]
> Hi @jasonblais the PR is now approved now but still, I can see the current status as "merge/blocked Expected — Waiting for status to be reported". Is there anything I have to do in regards to the PR?

@VishalSwarnkar Thanks! The PR actually undergoes two dev reviews and 1 QA review ([among the three reviews requested previously](https://github.com/mattermost/mattermost-webapp/pull/4611#event-2920941045)). You can learn more about our [code review process in our docs](https://developers.mattermost.com/contribute/getting-started/code-review/). Let me know if that's helpful.

As for the "merge/blocked Expected — Waiting for status to be reported" error, I'm not actually sure what that refers to, @saturninoabril would you know?

**Comment by saturninoabril** [comment]
@jasonblais I'm also not sure.
@VishalSwarnkar Once you make changes per comments, please update your branch into master and let's see what'll happen with the "merge/block".

**Comment by VishalSwarnkar** [comment]
Hi @saturninoabril as requested, I have incorporated the review comments and update the pull request. Kindly review

File: components/setting_picture.jsx [hunk 3]
```diff
@@ -335,7 +338,8 @@ export default class SettingPicture extends Component {
                                 />
                             </button>
                             <button
-                                tabIndex={disableSaveButtonFocus ? '-1' : '0'}
```
Comment (saturninoabril) [inline components/setting_picture.jsx]: Please revert this change which is for Accessibility.

Comment (VishalSwarnkar) [reply components/setting_picture.jsx]: @saturninoabril reverted the changes and updated the pull request

Comment (VishalSwarnkar) [reply components/setting_picture.jsx]: @saturninoabril reverted the changes
